beater/beater.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 type beater struct {
-	done   chan struct{}
+	errCh  chan error
```

Comment:
See my follow up comment. If we change the implementation we don't event need an error channel. WDYT?